packages/indexer/lib/index.js

```diff
@@ -198,116 +198,92 @@ class TransactionCollector {
     if (!lock && !type) {
       throw new Error("Either lock or type script must be provided!");
     }
-    if (lock) {
+    // Wrap the plain `Script` to `ScriptWrapper`.
+    if (lock && !lock.script) {
       validators.ValidateScript(lock);
+      this.lock = { script: lock, io_type: "both" };
+    }
+    if (lock && lock.script) {
```

Comment:
We can either execute line 202 - 205, or line 206 - 209, here we are doing script validation twice.

Reply:
You mean here doing condition check twice? The `ValidateScript` part would be executed only once.  I would use `else if` to concat those two seperate part.

Reply:
Think about this:

1. We provide a plain script here.
2. Line 202 - 205 will validate the script, then make it a script wrapper, and save it to `this.lock`
3. At line 206, the code finds that `this.lock` is a script wrapper, then enters the if
4. Line 207 validates the script in the script wrapper again.

Reply:
I don't think it will pass the second condition check if `this.lock` is set in line 202 - 205,  the `lock` in 206 is different thing from `this.lock` ?

Reply:
Oh yes, you are right, my mistake here.